validator/tests/unit3/test_scheduler/tests.py

```diff
@@ -181,6 +181,59 @@ def test_completion_on_finalize(self):
         with self.assertRaises(StopIteration):
             next(iterable)
 
+    def test_completion_on_last_result(self):
+        """Tests the that the schedule is not marked complete until the last
+        result is set.
+
+        Adds three batches with varying number of transactions, then tests
+        that they are returned in the appropriate order when using an iterator.
+        Test that the value of `complete` is false until the last value.
+
+        This test also finalizes the scheduler and verifies that StopIteration
+        is thrown by the iterator, and the complete is true in the at the end.
+        """
+        private_key = signing.generate_privkey()
+        public_key = signing.generate_pubkey(private_key)
+
+        context_manager = ContextManager(dict_database.DictDatabase())
```

Comment:
The context_manager now needs to be explicitly shutdown. In other scheduler tests there are setUp and tearDown methods doing that.